lib/main/pages/deferred_widget.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+
+typedef LibraryLoader = Future<void> Function();
+typedef DeferredWidgetBuilder = Widget Function();
+
+/// Wraps the child inside a deferred module loader.
+///
+/// The child is created and a single instance of the Widget is maintained in
+/// state as long as closure to create widget stays the same.
+///
+class DeferredWidget extends StatefulWidget {
```

Comment:
it is very cool, can you write an `ADR` to describe the solution, it will be a valuable documentation for other people can understand it in the future.

Reply:
Added.